lib/TPP/Dialect/Tpp/TppUtils.cpp

```diff
@@ -308,15 +308,49 @@ bool canMapToTppAdd(linalg::GenericOp linalgOp) {
   return hasOnlyScalarElementwiseOp<arith::AddFOp>(linalgOp.getRegion());
 }
 
-// TODO: check access pattern using affine map.
+// Return true if the linalgOp has only identity maps and its operands are
+// shapedType with rank <= 2.
+static bool checkOperandsAndMaps(linalg::GenericOp linalgOp) {
+  OpOperandVector operands(linalgOp.getDpsInputOperands());
+  operands.push_back(linalgOp.getDpsInitOperands()[0]);
+  for (OpOperand *operand : operands) {
+    AffineMap map = linalgOp.getMatchingIndexingMap(operand);
+    if (!map.isProjectedPermutation(/*allowZeroInResults=*/true))
+      return false;
+    ShapedType operandType = operand->get().getType().cast<ShapedType>();
+    // Reject 0-rank memref or memref with rank > 2.
+    if ((operandType.getRank() == 0))
+      return false;
+    ArrayRef<int64_t> shape = operandType.getShape();
+    size_t unitDims = 0;
+    for (int64_t dim : shape)
+      if (dim == 1)
+        unitDims++;
+    if (operandType.getRank() - unitDims > 2)
+      return false;
+  }
+  return true;
+}
+
+// Return true if the linalg.generic can be mapped to a tpp.add
+// 1. All loops are parallel
```

Comment:
It's best if these comments are inline in the function, before their implementation. Same below. Helps make sure that the comment is always in sync with the code and accurate.